support-workers/src/main/scala/com/gu/zuora/ProductSubscriptionBuilders.scala

```diff
@@ -157,8 +173,23 @@ trait ProductSubscriptionBuilder {
           Nil
         )
       ),
-      Subscription(contractEffectiveDate, contractAcceptanceDate, contractEffectiveDate, createdRequestId.toString, readerType = readerType)
+      Subscription(
+        contractEffectiveDate = contractEffectiveDate,
+        contractAcceptanceDate = contractAcceptanceDate,
+        termStartDate = contractEffectiveDate,
+        createdRequestId__c = createdRequestId.toString,
+        readerType = readerType,
+        autoRenew = autoRenew,
+        initialTerm = initialTerm,
+        initialTermPeriodType = initialTermPeriodType,
+      )
     )
+  }
+
+  def initialTermInDays(contractEffectiveDate: LocalDate, contractAcceptanceDate: LocalDate, termLengthMonths: Int): Int = {
+    val termEnd = contractAcceptanceDate.plusMonths(termLengthMonths)
+    Days.daysBetween(contractEffectiveDate, termEnd).getDays
```

Comment:
Maybe add comment, something like:
`// = Zuora's term length must accommodate the product's fixed term duration plus the lead time to the first paper date`